packages/schema-derive/src/query_responses.rs

```diff
@@ -1,23 +1,50 @@
-use syn::{parse_quote, Expr, ExprTuple, ItemEnum, ItemImpl, Type, Variant};
+use syn::{parse_quote, Expr, ExprTuple, Ident, ItemEnum, ItemImpl, Type, Variant};
 
 pub fn query_responses_derive_impl(input: ItemEnum) -> ItemImpl {
-    let ident = input.ident;
-    let mappings = input.variants.into_iter().map(parse_query);
-    let mut queries: Vec<_> = mappings.clone().map(|(q, _)| q).collect();
-    queries.sort();
-    let mappings = mappings.map(parse_tuple);
+    let is_nested = has_attr(&input, "query_responses", "nested");
 
-    // Handle generics if the type has any
-    let (impl_generics, type_generics, where_clause) = input.generics.split_for_impl();
+    if is_nested {
+        if !has_attr(&input, "serde", "untagged") {
```

Comment:
I didn't think about that. I added it, because not using untagged does not make sense with the semantics of the macro, but it will result in an error anyway when generating the schema, so I will remove the check.